lib/bunq/resource.rb

```diff
@@ -52,17 +54,34 @@ def append(path)
     end
 
     def ensure_session!
-      client.ensure_session!
+      client.current_session ||= client.session_servers.create
     end
 
     def with_session(&block)
-      client.with_session(&block)
+      ensure_session!
```

Comment:
`client.with_session` already ensures the session. So `with_session` can simply propagate to the client`